src/tox/session/env_select.py

```diff
@@ -101,11 +107,15 @@ class _ToxEnvInfo:
 
 
 class EnvSelector:
+
+    _warned_about: set[str]  #: shared set of skipped environments that were already warned about
+
     def __init__(self, state: State) -> None:
         # needs core to load the default tox environment list
         # to load the package environments of a run environments we need the run environment builder
         # to load labels we need core + the run environment
         self.on_empty_fallback_py = True
+        self._warned_about = set()
```

Comment:
You can move line 111 here; no need to define and then initialize it here; just initializing it is enough